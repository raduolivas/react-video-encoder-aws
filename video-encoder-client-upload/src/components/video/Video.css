.encoder-item-video {
    fload:left;
    margin:9px 7px 9px 0;
    display:inline-block;
    cursor: pointer;
    color:#FFFFFF;
}
.encoder-item-video
.encoder-video-thumb {
    width:143px;
    height:94px;
    background-color: #FFFFFF;
    padding:6px 6px;
}

.encoder-item-video
.encoder-video-thumb
.encoder-video-status {
    position:relative;
}

.encoder-video-status span {
    font-size: 11px;
}