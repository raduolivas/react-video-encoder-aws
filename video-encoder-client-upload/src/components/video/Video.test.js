import React from 'react';
import { configure, shallow, mount } from 'enzyme';
import ShallowRenderer from 'react-test-renderer/shallow';
import Adapter from 'enzyme-adapter-react-16';
import { stub } from 'sinon';



import mock from '../../mocks/single-video-mock.json';

import Video from './';

configure({adapter: new Adapter()});

describe('Video ', () => {
    let wrapper;
    let props;

    beforeEach(() => {

        props = {
            video: mock,
            key: undefined,
            clicked: undefined
        }
        wrapper = shallow(<Video {...props}/>);
    })

    it("always renders a div", () => {
        expect(wrapper.find('div').length).toBeGreaterThan(0);
    });

    it('receive clicked props', () => {
        const render = new ShallowRenderer();
        render.render(<Video clicked={()=>{}} />);
        const tree = render.getRenderOutput();
        expect(typeof wrapper.props().clicked === 'function');
        expect(tree).toMatchSnapshot();
    });

    it('receive video props', () => {
        wrapper = shallow(<Video video={mock}/>)
        const render = new ShallowRenderer();
        // render.render(<Video video={mock} />);
        const tree = render.getRenderOutput();
        expect(tree).toMatchSnapshot();
    });

})