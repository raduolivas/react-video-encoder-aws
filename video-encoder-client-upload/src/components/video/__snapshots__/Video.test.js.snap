// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video  receive video props 1`] = `null`;
